### ISSUE TITLE
1. ASNHC:Mamm - Manage Collection

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/data-migration#72]
Follow instructions at [How to Manage a Collection](http://handbook.arctosdb.org/how_to/How-to-Manage-a-Collection-in-Arctos.html)

**Comment by Jegelewicz** [comment]
> Dianna,
> 
> I thought that Joey was asking about Arctos. You have to give him permissions to work with your Arctos data. We should probably take a step back and make sure that you have manage collection access and have your collection set up properly. See Step one in your Data Migration GitHub project. Let me know if you have questions about the manage collection form.

Teresa J. Mayfield-Meyer

**Comment by diannakrejsa** [comment]
![Screen Shot 2019-07-30 at 9 41 07 AM](https://user-images.githubusercontent.com/36166607/62139159-7811d900-b2ae-11e9-9cda-24758646d89f.png)

Not sure if I've been given full operational permissions to invite users to become Operators. When I hover over Manage Arctos I only have 'Developer Tools' available, not Manage Arctos > Roles/Permisions etc.

**Comment by Jegelewicz** [comment]
Dianna,

You are correct. Please read [Sharing Data and Resources](http://handbook.arctosdb.org/documentation/sharing-data-and-resources.html). I will give you access to add your operators.

**Comment by diannakrejsa** [comment]
Hi Teresa, 

I was attempting to give the ASU App Team user account "PUBLIC" and "COLDFUSION_USER" roles and got the following error: 
<img width="1289" alt="Screen Shot 2019-07-30 at 2 31 56 PM" src="https://user-images.githubusercontent.com/36166607/62159313-fda97f00-b2d6-11e9-9c5b-547670227ff8.png">

I also attempted to simply "Invite as Operator" and got the following error. Is there something up with this user/log-in or something up with adding roles in Arctos?

<img width="1288" alt="Screen Shot 2019-07-30 at 2 32 17 PM" src="https://user-images.githubusercontent.com/36166607/62159367-29c50000-b2d7-11e9-989e-bc42918fefd5.png">

Thanks!

**Comment by Jegelewicz** [comment]
Dianna,

Before giving anyone roles, their username must be associated with an agent. Take a look at [Creating your Arctos Team](http://handbook.arctosdb.org/how_to/How-to-Create-your-Arctos-Team-Users-and-Operators.html) to be sure you follow all of the steps. If that doesn't work, let me know!

**Comment by Jegelewicz** [comment]
Also see #71

**Comment by diannakrejsa** [comment]
I think I did follow all the steps for How to Create and Manage Your Arctos Team, I ran through it yesterday and "ASU App Team" should be an agent and listed as technical support (or whatever is on the picklist) for ASNHC Mammals. However, I now tried to navigate to the initial Arctos page to run through the steps of the How to Create and Manage Your Arctos Team again and I got this bounce back message:
<img width="1278" alt="Screen Shot 2019-07-31 at 9 38 24 AM" src="https://user-images.githubusercontent.com/36166607/62221379-31d47c80-b377-11e9-849a-47cab88759fa.png">

**Comment by diannakrejsa** [comment]
Yeah, I tried it again (after getting myself removed from the "blacklist," lol) and I get the same errors. I think this "ASU App Team" is set up fine as a user and Agent, I just can't add them as an Operator. Thoughts?

**Comment by dustymc** [comment]
If you will let me know your IP I can investigate that; there's likely a compromised machine on your network.

First and foremost, you should consider NOT creating group accounts. I think that has been done and there may still be a way to sneak it through, but it is not something Arctos is designed for and it will lead to mystery-data.

There is no user account - you can search here...

<img width="301" alt="Screen Shot 2019-07-31 at 8 05 47 AM" src="https://user-images.githubusercontent.com/5720791/62223605-17e06d00-b36a-11e9-8503-e9bb27936889.png">

... that corresponds with the 'login' name in https://arctos.database.museum/agents.cfm?agent_id=21317068

I've updated http://handbook.arctosdb.org/documentation/notifications.html to better explain notifications, and http://handbook.arctosdb.org/documentation/users.html contains Operator requirements.

**Comment by Jegelewicz** [comment]
> First and foremost, you should consider NOT creating group accounts. I think that has been done and there may still be a way to sneak it through, but it is not something Arctos is designed for and it will lead to mystery-data.

I totally agree. Both Colegate and Joey should have individual user accounts and be invited individually as operators. @diannakrejsa let me know if you want to set up a call about this.

**Comment by diannakrejsa** [comment]
I was searching for "ASU App Team" and that's where I was finding the agent information linked, etc. I wasn't using asu_app_team if that makes a difference.

At any rate, I suppose it doesn't matter because I'll have Joey create his own personal log-in (user name, Agent, etc.) and then try adding Operator to that. 

IP address: 72.18.174.43

**Comment by snknts** [comment]
Joey here :)
I'm getting this error when confirming my password to become an operator:
![Screen Shot 2019-08-06 at 2 22 56 PM](https://user-images.githubusercontent.com/53309704/62571000-dbbf7780-b855-11e9-8aca-0a4652991b4e.png)

**Comment by campmlc** [comment]
Hi Joey,
what is your username?

**Comment by snknts** [comment]
jherefor@angelo.edu
sorry, thought you could look that up... but I forgot that arctos != github :P

**Comment by snknts** [comment]
Still getting that error.. is there something I'm doing wrong?
(Or a step we're skipping?)

**Comment by Jegelewicz** [comment]
@snknts it could be your password? Make sure it adheres to this:

> Passwords should be at least eight characters, start with a letter, contain only A-Z, a-z, 0-9, and !$%&()`*+,-/:;?_., not contain the username, and consist of at least one letter, number, and special character

**Comment by snknts** [comment]
It did take me a while to get over that hurdle.
(The setting my password requirements and validating my password on that form use different requirements it seems)

I eventually got it to validate though (then that error appears)

**Comment by Jegelewicz** [comment]
@dustymc I'm at a loss on this one - any ideas?

**Comment by dustymc** [comment]
Please see http://handbook.arctosdb.org/documentation/users.html#creating-operators

>account name must use only letters and numbers (ASCII characters A-Z, a-z, and 0-9), and cannot begin with a number.


That's linked at READ THIS FIRST! here:

<img width="227" alt="Screen Shot 2019-08-09 at 10 18 16 AM" src="https://user-images.githubusercontent.com/5720791/62796864-1e1fc900-ba8f-11e9-9f4d-e3da2feb4121.png">

I'm not sure what else we could do with that??

**Comment by Jegelewicz** [comment]
@snknts and @diannakrejsa see above. Joey is going to need to create a new username and start the process over.

@dustymc as @campmlc and I have said before, these rules should apply to non-operators as well. It would avoid this sort of stuff from the get-go.

**Comment by dustymc** [comment]
>should apply to non-operators as well

Issue please - seems overly burdensome to the ~90% of users who aren't operators to me, but I suppose I'm up for anything!

**Comment by campmlc** [comment]
Agree, this is causing continual issues with new users, so best make it
easy for our operators at the cost of making it slightly harder for the
public.

**Comment by snknts** [comment]
I'll give that a shot.

**Comment by snknts** [comment]
have a new issue now... odds are I'm doing something wrong... but I'm not sure what.. options seem limited... so It doesn't seem like I could mess it up..

I created a new account (Joey76909) 
but... 
Dianna can't find me on the agent search (to begin the process to make us operators)
any idea what I did wrong?  Do I need to create it differently somehow?  

Do I need to delete my old user (with the invalid operator names)?  how do I do that if so?

**Comment by Jegelewicz** [comment]
Dianna needs to change your Agent profile and add your new login id, then re-invite you to be an operator using that username.

![image](https://user-images.githubusercontent.com/5725767/62980703-826ebf80-bde4-11e9-9348-5133e34c4261.png)

**Comment by Jegelewicz** [comment]
> Teresa,
> 
> I hope you are doing well.  Our team is ramped up and helping work through our prep for migration of our data into Arctos.  
> 
> Our team has been going through a good number of steps via GetHub to get the necessary access to Arctos.  We are seeing lots of back and forth on this and it is actually holding up some of our next steps in our timeline.  
> 
> We are needing to get Joey Hereford and Colegate Spinks setup correctly in Arctos to allow us to help Dianna and Team in the right ways.  
> 
> Can you give Dianna or me a call to discuss and help us get this knocked out?  We would sure appreciate your help and if there is someone else that we should talk to, please let us know. 
> 
> Dianna’s Cell - 515-979-4137
> Brian J. - 325-374-0814
> 
> 
> Thank you for your help,

Brian Jackson
Associate Director, Information Technology
Angelo State University
Member, Texas Tech University System
ASU Station #11020
San Angelo, TX 76909-1020
Phone: (325) 942-2333 Cell: (325) 374-0814 Fax: (325) 942-2382
Brian.Jackson@angelo.edu

**Comment by Jegelewicz** [comment]
I left a message for @diannakrejsa I can call tomorrow except between 10:30am and 11:30am MDT.

**Comment by diannakrejsa** [comment]
Hey Teresa!


Thanks for your message, for some reason my cell phone is not allowing me to make calls and it appears this started about 5pm (which is why I missed yours!). At any rate, I changed Joey's original agent profile login to the new username and sent the Operator invitation. I thought it might have to involve creating a whole new agent for him, I'm glad it was that simple.


The ASU team can proceed from here with adding public and coldfusion access to Joey's profile once he's accepted the invitation, but if you think it'd be good to have a group chat perhaps with a shared screen to launch further ahead on a number of other things, perhaps we can arrange a group call. My schedule is fairly flexible, but others may already be booked. We always meet Tuesdays at 2pm CT/1pm MDT, so maybe we can schedule that now if something sooner doesn't work on our end.


Joey/Bryan/Colegate, what is your availability in the next day or two for an Arctos walk-through and getting our operator powers fully up and running?


Thanks very much,

Dianna


[ASU]
Dianna M. Krejsa
Collections Manager, Angelo State Natural History Collections
Angelo State University
Member, Texas Tech University System
ASU Station #10890
San Angelo, TX 76909-0890
Phone: (325) 486-6699, Cell: (515) 979-4137
Office: Cavness 015
dkrejsa@angelo.edu
<mailto:robert.dowler@angelo.edu>
________________________________

**Comment by Jegelewicz** [comment]
Dianna,

We should definitely talk soon. I am concerned that you guys are jumping ahead without reading documentation or consulting with anyone and I don't want you to get frustrated or cause community issues. I am available today after noon MDT and all day tomorrow. Just let me know when and what number to call or we can use the Arctos Zoom room to conference.

**Comment by diannakrejsa** [comment]
How about tomorrow morning? I am available before 11am MDT. I'm not sure if Joey, Brian, or Colegate can be available but I can take notes. Are there specific topics we need to cover first, either from your perspective Teresa, or yours Joey?

**Comment by campmlc** [comment]
Hi Dianna,
It would be great if all your potential users could attend the meeting? I
can do 8am or 10 am MDT.
Mariel

**Comment by snknts** [comment]
tomorrow morning works for me :)

**Comment by diannakrejsa** [comment]
Let's do 10am MDT/11am CT. We can meet at my office Joey?

**Comment by snknts** [comment]
that works :)

(Do we need a webex setup?)

**Comment by campmlc** [comment]
Here you go:
Join from PC, Mac, Linux, iOS or Android: https://berkeley.zoom
.us/j/363481382

Or iPhone one-tap :
    US: +16465588656,,363481382#  or +16699006833,,363481382#
Or Telephone:
    Dial(for higher quality, dial a number based on your current location):
        US: +1 646 558 8656  or +1 669 900 6833  or +1 877 369 0926 (Toll
Free) or +1 877 853 5247  (Toll Free)
    Meeting ID: 363 481 382
    International numbers available: https://zoom.us/u/ntnS3jsn

**Comment by diannakrejsa** [comment]
Thanks!

**Comment by diannakrejsa** [comment]
Hi folks, 

I'm attempting an short upload to make sure the bulkloader column headings will work before a larger attempt. I am correct in reading that the error for this particular CSV upload is that the "PART_CONTAINER_LABEL_4" is the single error as an invalid identifier? I don't know what's wrong with this column heading, I copied-pasted the "PART_CONTAINER_LABEL_3" and changed the 3 to a 4 to see if that would correct it, which didn't work. Kind of stuck, is there something I'm missing you all can point out easily? 

<img width="2244" alt="Screen Shot 2019-08-20 at 11 57 38 AM" src="https://user-images.githubusercontent.com/36166607/63367610-daa44500-c341-11e9-8fbd-8157e53f8d82.png">

[ASNHC 18737-18771.xlsx](https://github.com/ArctosDB/data-migration/files/3521743/ASNHC.18737-18771.xlsx)

**Comment by dustymc** [comment]
The bulkloader builder...

<img width="318" alt="Screen Shot 2019-08-20 at 10 11 48 AM" src="https://user-images.githubusercontent.com/5720791/63368530-f48a5b80-c332-11e9-830c-d60f6d29c17b.png">


is the one and only authoritative source of column names (because it pulls directly from the target table). PART_CONTAINER_LABEL_N was removed some time ago.

That sort of error - a structural conflict - stops the entire process. It's not necessarily the only problem, the loader can't see past it to know.

**Comment by diannakrejsa** [comment]
I've temporarily left the 'Bulkload Specimen Data' issue (I'll be doing that once I hit submit on this comment) to see if I can finish getting tissue locations associated with a previous bulkload of specimen data. I am doing this through a bulkload of specimen part attributes (https://arctos.database.museum/tools/BulkloadAttributes.cfm?action=managemystuff). I have done this once before with a CSV Teresa put together for me. In an effort to avoid errors and update the column headings for Arctos bulkloading, I'm figuring out the meaning of the new or different column headings. Should I reference the Bulkloader Builder again, or reflect the column headings found at the bulkload specimen part attributes window? Is there a list of definitions for these headings somewhere? Attached is the data I'm trying to get in, green highlighted column headings have been matched up, blues are new and not sure what I need to populate them with, white are important and need to be mapped to correct column headings.

Also, is there a way to bulkload a csv of all specimen parts for a set of catalog numbers at once, or does it have to be submitted one part per GUID at a time? E.g., if ASNHC:Mamm:16530 has two tissue tubes, do I have to submit the attribute part locations in two separate CSVs, or is there a way to do both at once? I believe we discussed before that they had to be bulkloaded in separate CSVs.

[BulkloadSpecimenPartAtt (Part 2).xlsx](https://github.com/ArctosDB/data-migration/files/3526691/BulkloadSpecimenPartAtt.Part.2.xlsx)

**Comment by Jegelewicz** [comment]
@diannakrejsa I am swamped today but I will look at this as soon as I can!

**Comment by dustymc** [comment]
>Bulkloader Builder again, or reflect the column headings found at the bulkload specimen part attributes window

The latter - the Builder is ONLY for the specimen bulkloader, all component loaders will have a template you can download. (But generally not customize because they're already small.)

There's a link to http://arctos.database.museum/info/ctDocumentation.cfm?table=CTSPECPART_ATTRIBUTE_TYPE which is the definitive attribute documentation. Let me know if anything else is missing and I'll try to get at least skeletal documentation up.

The loader will handle any number of attributes for any number of parts for a specimen. The limitation is that it must be able to uniquely identify a part. So if ASNHC:Mamm:16530 has two "tissue" (or anything else) parts, the loader will fail. (Barcodes are a convenient way to uniquely identify parts). If ASNHC:Mamm:16530 has a "tissue" and a "something else" then there should not be a problem.

If you do have multiple identical parts and some sort of mechanism for selecting one of them, I can help SQL the data in.

**Comment by diannakrejsa** [comment]
Great! Thanks for the answers. Only occasionally do parts double up with two tubes of "tissue" or "muscle" on a single GUID, so screening for only that particular situation should make things simpler when we're adding part attributes. 

Okay, back to the Specimen Bulkloader. I updated all the columns and things are relatively happy. The last two errors involve TAXON_NAME--Damaliscus pygargus phillipsi not found. I think we have the correct subspecies on this, so does it need to be added to taxon tables?

I had been having an error with HIGHER_GEOG with "Africa, South Africa, Eastern Cape" because it needed to be "Africa, South Africa, Eastern Cape Province." Some other provinces in South Africa (as well as other provinces and states in general) don't require that suffix in Arctos, while others do require it within Arctos. I can modify all our records to have the "Province" at the end for Eastern Cape, but just wondering if there is a geographic rule behind this.

[ASNHC 18737-18771 bulkloader_stage.xlsx](https://github.com/ArctosDB/data-migration/files/3527423/ASNHC.18737-18771.bulkloader_stage.xlsx)

**Comment by dustymc** [comment]
>TAXON_NAME

That passes "the Google test" so yes, it can be added.

https://www.google.com/search?q=%22Damaliscus+pygargus+phillipsi%22

>South Africa

Please file an Issue for that sort of thing - https://github.com/ArctosDB/arctos/issues/2227

Geography SHOULD follow the rules at http://handbook.arctosdb.org/documentation/higher-geography.html#guidelines-for-geographic-terms-in-arctos, but reality...

This...

<img width="406" alt="Screen Shot 2019-08-21 at 2 52 26 PM" src="https://user-images.githubusercontent.com/5720791/63471190-588c4d00-c423-11e9-9f72-8cb5c88cbe90.png">

... may be helpful.

**Comment by Jegelewicz** [comment]
Damaliscus pygargus phillipsi has been added.

**Comment by Jegelewicz** [comment]
@dustymc I can't change any geog that has been used, correct? Could you make all of these states include the designation "Province" so that they are consistent?

![image](https://user-images.githubusercontent.com/5725767/63472535-b161e280-c42f-11e9-8f0b-ad3db1ecb356.png)

**Comment by dustymc** [comment]
I think it needs a person involved. Transvaal should probably just be Transvaal, I think, and I can't edit it without a wiki reference anyway.

Geography is editable - there's no clear right-and-wrong like taxonomy, and corrections (like this) are often necessary. Changes will go out to everyone in the authority notification.

**Comment by Jegelewicz** [comment]
@diannakrejsa short answer for now is put "Province" on all of your "Africa, South Africa, Eastern Cape" so they will load. Eventually the geography committee, or someone else with the free time will clean up the inconsistencies.

**Comment by diannakrejsa** [comment]
Sounds good, that's what I did and got that small batch to load.

**Comment by Jegelewicz** [comment]
Clean up requested and challenge accepted - https://github.com/ArctosDB/arctos/issues/2227

**Comment by diannakrejsa** [comment]
I have a data conformity question--we have a large number of specimens from zoos (captive instead of wild caught for COLLECTING_SOURCE). Historically the mammals curator doesn't list higher geography for these individuals, because of the implication that they are genetically from "North America, United States, Texas, Bexar County" when they really just died at the San Antonio Zoo. 

How are records like this in other collections? I know HIGHER_GEOG is a required field with controlled vocabulary. Is there a dummy HIGHER_GEOG equivalent to "unknown" or "captive colony" or are we required to provide that string to Bexar County?

**Comment by campmlc** [comment]
We put the higher geography for the zoo locality, but the captive flag has
previously prevented these records from mapping. However, there is an open
discussion on whether this is necessary. I can't access the issue from my
phone, will try later.

**Comment by dustymc** [comment]
That issue is https://github.com/ArctosDB/arctos/issues/2068

PLEASE include those data, and we'll figure out how to display them (or not) as a community in the issue. Anything else is actively prohibiting scientific inquiry. Do ostriches that lived in Florida have different lung problems than ostriches that lived in Minnesota? Beats me, we're withholding those data...
